src/cli/cli_config.h

```diff
@@ -56,4 +56,30 @@
 #define OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH 384
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_ENABLE
+ *
+ * Define as 1 for CLI to emit its command input string and the resulting output to the logs.
+ *
+ * By default this is enabled on any POSIX based platform (`OPENTHREAD_POSIX`) and only when CLI itself is not being
+ * used for logging.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_ENABLE
+#define OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_ENABLE \
+    (OPENTHREAD_POSIX && (OPENTHREAD_CONFIG_LOG_OUTPUT != OPENTHREAD_CONFIG_LOG_OUTPUT_APP))
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_LOG_STRING_SIZE
+ *
+ * The log string buffer size (in bytes).
+ *
+ * This is only used when `OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_ENABLE` is enabled.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_LOG_STRING_SIZE
+#define OPENTHREAD_CONFIG_CLI_LOG_INPUT_OUTPUT_LOG_STRING_SIZE 200
```

Comment:
Why 200 ? Why not use `OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH` ?

Reply:
Changed in new push.